### PR TITLE
Ondemand audio (Radio and Podcasts) - Remove tests (Unit, integration and E2E) against AMP versions of these pages

### PR DESCRIPTION
**Pull request by LilyL0u**
Resolves JIRA [number]

Overall changes
======
_A very high-level summary of easily-reproducible changes that can be understood by non-devs, and why these changes where made._

Code changes
======

- _A bullet point list of key code changes that have been made._

Testing
======
1. _List the steps used to test this PR._

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/integration/common/radioSchedule.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const servicesWithRadioSchedule = ['indonesia'];
 const servicesWithoutRadioSchedule = ['gahuza'];
 
-export default ({ isAmp }) => {
+export default ({ isAmp = false }) => {
```
Comment (karinathomasbbc) [inline src/integration/common/radioSchedule.js]: I think we should revert this change - I believe this test is called from other pages which may or may not have a radio schedule, and defaulting the value to false might still be required.

If it's not used from anywhere, let's remove the `isAmp` param and all references altogether.

File: src/integration/pages/onDemandAudioPage/indonesia/canonical.test.js [hunk 2]
```diff
@@ -4,11 +4,9 @@
  */
 
 import runCrossPlatformTests from '../crossPlatformTests';
-import runCanonicalTests from '../canonicalTests';
```
Comment (karinathomasbbc) [inline src/integration/pages/onDemandAudioPage/indonesia/canonical.test.js:7]: Did we mean to delete these?

File: src/integration/pages/onDemandAudioPage/crossPlatformTests.js
Comment (karinathomasbbc) [inline src/integration/pages/onDemandAudioPage/crossPlatformTests.js:1]: I think we should move all tests to the canonical tests file instead of here - for me, `crossPlatform` is confusing. In this case, we don't have crossPlatform tests, but for other page types we do, so let's be explicit.

File: src/integration/pages/onDemandAudioPage/indonesia/canonical.test.js (hunk 2 above)
Comment (LilyL0u) [reply src/integration/pages/onDemandAudioPage/indonesia/canonical.test.js:7]: I have moved all the canonical tests into the crossPlatform tests as we only need one file now instead of amp, canonical and crossPlatform. It could have been done either way -  canonical into crossPlatform or crossPlatform into canonical, as long as they were consolidated into one file. See comment below about the plan to rename.

File: src/integration/pages/onDemandAudioPage/crossPlatformTests.js
Comment (LilyL0u) [reply src/integration/pages/onDemandAudioPage/crossPlatformTests.js:1]: I was planning on changing the name crossPlatform to something else, and was going to ask people what they thought would be a good name. Could just be tests.js? I would also change all the canonical.test.js file names in each page folder. Again could just be tests.js?


I can do it the other way round if preferred to keep the name canonical.

File: src/app/routes/utils/regex/utils/index.js [hunk 3]
```diff
@@ -146,7 +146,7 @@ export const getPodcastBrandRegex = services => {
 
 export const getOnDemandRadioRegex = services => {
   const serviceRegex = getServiceRegex(services);
-  return `/:service(${serviceRegex}):variant(${variantRegex})?/:serviceId(${radioMasterBrandRegex})(/programmes)?/:mediaId(${mediaIdRegex}):amp(${ampRegex})?:lite(${liteRegex})?`;
```
Comment (LilyL0u) [inline src/app/routes/utils/regex/utils/index.js:149]: I realised we also have these tests for valid route regex, including amp routes. Do you think we should also change these in these removing tests PRs @karinathomasbbc ? There are tests for this in the routes/utils/regex file.

If so, I can also do this in the OD TV PR.

File: src/integration/common/radioSchedule.js (hunk 1 above)
Comment (LilyL0u) [reply src/integration/common/radioSchedule.js]: I will remove the default false value, but leave isAmp as your belief is correct and this test is also used in the live radio pages. I can remove it in the live radio PR if we merge them in that order.

File: src/app/routes/utils/regex/utils/index.js (hunk 3 above)
Comment (karinathomasbbc) [reply src/app/routes/utils/regex/utils/index.js:149]: Given that we're making these changes here, it means that we must release the Belfrage PR first, and have the .amp page 404 at the source, rather than 404ing in Simorgh.

So yes, I think we should make the corresponding change in the OD TV PR for consistency (but ideally, this probably should be part of the final cleanup / removal of all AMP code e.g. as part of the ticket to remove embedUrl)

File: src/integration/pages/onDemandAudioPage/crossPlatformTests.js [hunk 4]
```diff
@@ -10,4 +13,7 @@ export default service => {
   runCommonEpisodeTests();
   runRecentEpisodesTests();
   runMediaPlayerEmbedLegacyTests();
+  runCoreCanonicalTests();
+  runCanonicalAnalyticsTests();
+  runRadioScheduleTests({ isAmp: false }); // keep this for now until live radio also has amp removed
```
Comment (karinathomasbbc) [inline src/integration/pages/onDemandAudioPage/crossPlatformTests.js]: `isAmp` is false by default, so let's remove it

```suggestion
  runRadioScheduleTests();
```

Comment (karinathomasbbc) [reply src/integration/pages/onDemandAudioPage/crossPlatformTests.js]: Also - not sure if I mentioned this already or not. I think we should move the crossPlatform tests to the canonical file - since onDemandAudio is no longer cross platform, this is confusing.

File: src/integration/pages/onDemandAudioPage/indonesia/canonical.test.js (hunk 2 above)
Comment (karinathomasbbc) [reply src/integration/pages/onDemandAudioPage/indonesia/canonical.test.js:7]: I think we should move crossPlatform into canonical for consistency & to prevent confusion

**Review by karinathomasbbc — CHANGES_REQUESTED** [review]

File: src/integration/pages/onDemandAudioPage/crossPlatformTests.js (hunk 4 above)
Comment (LilyL0u) [reply src/integration/pages/onDemandAudioPage/crossPlatformTests.js]: I was thinking we would change the name of the crossPlatform test file anyway, as yes it doesn't mean anything anymore. But also the 'canonical' distinction doesn't mean anything now either, so I think even using the word canonical is maybe puzzling. In the case of me changing the file name, I don't think I need to change any of the other code as they have now been consolidated into one file, and we can just choose what to call that file. I may have missed something though that you have seen related to consolidating the files in the different direction.

Comment (LilyL0u) [reply src/integration/pages/onDemandAudioPage/crossPlatformTests.js]: The tests fail if I remove this without providing the default false value, but you suggested reverting that in the comment https://github.com/bbc/simorgh/pull/11799#discussion_r1692955171, although I may have misunderstood the reasoning

Comment (karinathomasbbc) [reply src/integration/pages/onDemandAudioPage/crossPlatformTests.js]: An optional param like `isAmp` should always have a default. So if it is defaulted to `false` then it means we don't need to include it. Hopefully reverting the removal of the default value will fix the test - worth a shot?

File: src/integration/common/radioSchedule.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 const servicesWithRadioSchedule = ['indonesia'];
 const servicesWithoutRadioSchedule = ['gahuza'];
 
-export default ({ isAmp }) => {
+export default ({ isAmp = false } = {}) => {
```
Comment (karinathomasbbc) [inline src/integration/common/radioSchedule.js:4]: Did we need the extra `= {}`
```suggestion
export default ({ isAmp = false }) => {
```

**Review by karinathomasbbc — APPROVED** [review]

File: src/integration/common/radioSchedule.js (hunk 5 above)
Comment (LilyL0u) [reply src/integration/common/radioSchedule.js:4]: If I do that I get the error 'TypeError: Cannot read properties of undefined (reading 'isAmp')'.
Having the extra = {} stops this error.

Comment (LilyL0u) [reply src/integration/common/radioSchedule.js:4]: Apparently this is because {isAmp = false} as the default value then expects an object being passed in. If we pass in nothing with radioSchedule(), then it is getting 'undefined' which is trying to be destructured, but cannot as it isn't an object. Adding = {} provides an empty object as the default value for the entire parameter, so when nothing is provided in radioSchedule() call it defaults to an empty object, which then when destructuring happens it becomes the default false, so does not error.

Comment (karinathomasbbc) [reply src/integration/common/radioSchedule.js:4]: Ahhhh I see. It probably doesn't need to be a destructed object anymore (i.e. it could be a single param with a default value), but let's leave it for now, thanks for checking! 👍

**Review by amoore108 — APPROVED** [review]
